openapi3/swagger_loader.go

```diff
@@ -30,10 +30,11 @@ type SwaggerLoader struct {
 	Context                context.Context
 	LoadSwaggerFromURIFunc func(loader *SwaggerLoader, url *url.URL) (*Swagger, error)
 	visited                map[interface{}]struct{}
+	loadedRemoteSchemas    map[url.URL]*Swagger
 }
 
 func NewSwaggerLoader() *SwaggerLoader {
-	return &SwaggerLoader{}
+	return &SwaggerLoader{loadedRemoteSchemas: map[url.URL]*Swagger{}}
```

Comment:
Note that since this is only initialized/cleared once at creation there is the possibility that you could introduce unwanted behavior for existing users. For example, if you re-use the loader instance to reload an OpenAPI doc and expect to pull in the updated references as well this will fail after the proposed change.

I just checked and this is *not* a problem for [API Sprout since it always creates a new loader](https://github.com/danielgtaylor/apisprout/blob/master/apisprout.go#L321). Just calling out that it might be problematic for others.

Reply:
Could you mention this in the docs (for example as the documentation for this very function?). It's better than buried in a PR comments :)

Also, could you not initialize the map here but instead check for `nil` in `resolveComponent` below? This would save allocation in the case the user does not rely on remote schemas.

Reply:
Ah well @danielgtaylor you're not the author... Thanks for pointing this out!